static/js/lib/redux_api_test.js

```diff
@@ -0,0 +1,341 @@
+          let result = reducer[action.successType](
+            {}, { type: 'ACTION', payload: { some: 'DATA' }}
+          );
+          assert.deepEqual(result, {
+            [`${R.toLower(verb)}Status`]: FETCH_SUCCESS,
+            loaded: true,
+            processing: false,
+            data: { some: 'DATA' }
+          });
+        });
+      });
+
+      it('should represent failure', () => {
+        checkForVerbs(endpoint, verb => {
+          let action = actions[R.toLower(verb)];
+          let reducer = deriveReducer(endpoint, action, verb);
+          let result = reducer[action.failureType](
+            {}, { type: 'ACTION', payload: { some: 'ERROR' }}
+          );
+          assert.deepEqual(result, {
+            [`${R.toLower(verb)}Status`]: FETCH_FAILURE,
+            loaded: true,
+            processing: false,
+            error: { some: 'ERROR' }
+          });
+        });
+      });
+    });
+
+    describe('deriveReducers', () => {
+      it('should clear itself', () => {
+        let reducer = deriveReducers(endpoint, actions);
+        let result = reducer({}, { type: actions.clearType });
+        assert.deepEqual(result, INITIAL_STATE);
+      });
+
+      it('should use an INITIAL_STATE prop on the Endpoint, if present', () => {
+        endpoint.initialState = {
+          extraProp: 'HI',
+          wow: 'yeah...'
+        };
+
+        let reducer = deriveReducers(endpoint, actions);
+        let result = reducer({}, { type: actions.clearType });
+        assert.deepEqual(result, endpoint.initialState);
+      });
+
+      it('should return a function', () => {
+        let reducer = deriveReducers(endpoint, actions);
+        assert.isFunction(reducer);
+      });
+
+      it('should return a different state for all defined action types', () => {
+        let reducer = deriveReducers(endpoint, actions);
+
+        checkForVerbs(endpoint, verb => {
+          let action = actions[R.toLower(verb)];
+          checkForActionTypes(action, type => {
+            let result = reducer({}, { type: type, payload: 'SOME_DATA' });
+            assert.notDeepEqual(result, {});
+          });
+        });
+      });
+
+      it('should include any extraActions', () => {
+        endpoint.extraActions = {
+          'MY_NEW_ACTION_TYPE': () => ({ not: 'legit'})
+        };
+
+        let reducer = deriveReducers(endpoint, actions);
+        let result = reducer({}, { type: 'MY_NEW_ACTION_TYPE' });
+        assert.deepEqual(result, { not: 'legit' });
+      });
+
+      it('should no-op if reducer for an action type is not present', () => {
+        let reducer = deriveReducers(endpoint, actions);
+
+        let initialState = { initial: 'State' };
+        let result = reducer(initialState, { type: '❤❤❤❤❤', payload: '<3<3<3<3<3'});
+        assert.equal(result, initialState);
+      });
+    });
+
+    describe.skip('exported reducers', () => { // eslint-disable-line mocha/no-skipped-tests
```

Comment:
Yes :) I just wanted to not have to re-write the tests when I do the follow-up PR to switch over the coupons reducer.